cli/command/service/ps.go

```diff
@@ -43,13 +47,31 @@ func runPS(dockerCli *command.DockerCli, opts psOptions) error {
 	client := dockerCli.Client()
 	ctx := context.Background()
 
-	service, _, err := client.ServiceInspectWithRaw(ctx, opts.serviceID)
-	if err != nil {
-		return err
+	filter := opts.filter.Value()
+	for _, service := range opts.services {
+		serviceIDCount := 0
+		serviceIDFilter := filters.NewArgs()
+		serviceIDFilter.Add("id", service)
+		if serviceList, err := client.ServiceList(ctx, types.ServiceListOptions{Filter: serviceIDFilter}); err == nil {
+			for _, serviceEntry := range serviceList {
+				filter.Add("service", serviceEntry.ID)
+				serviceIDCount++
+			}
+		}
+		serviceNameFilter := filters.NewArgs()
+		serviceNameFilter.Add("name", service)
+		if serviceList, err := client.ServiceList(ctx, types.ServiceListOptions{Filter: serviceNameFilter}); err == nil {
```

Comment:
@dnephin It already happens with `docker node ps`.

Not too worried about making extra calls, because the user will do it anyway. But if we can optimize it all the better.